api/server/httputils/errors.go

```diff
@@ -43,11 +43,11 @@ func GetHTTPErrorStatusCode(err error) int {
 		statusCode = http.StatusServiceUnavailable
 	case errdefs.IsForbidden(err):
 		statusCode = http.StatusForbidden
-	case errdefs.IsNotModified(err):
+	case errdefs.IsNotModified(err) || errdefs.IsAlreadyExists(err):
```

Comment:
`IsAlreadyExist()` seems like it could mean not modified, but in other contexts it could mean Conflict.

Reply:
Agreed, but in that scenario it should be up to the thing that received the error to determine if it's a hard failure like a conflict or already exists.
The doc string for `ErrAlreadyExists` spells out that this is a special case of `ErrNotModified`